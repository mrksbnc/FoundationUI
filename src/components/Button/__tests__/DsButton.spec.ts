import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';
import { ButtonSize, ButtonType, DsButton } from '..';

const btnType = ButtonType.primary;
const btnSize = ButtonSize.base;

describe('DsButton', () => {
	test('renders correctly', () => {
		const wrapper = mount(DsButton, {
			props: {
				label: 'Test',
				size: 'small' as ButtonSize,
				type: 'primary' as ButtonType,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
