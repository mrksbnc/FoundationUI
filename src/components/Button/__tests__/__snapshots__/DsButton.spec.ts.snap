// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DsButton > renders correctly 1`] = `
"<button id=\\"Ds-button_1692481095537\\" class=\\"Ds-button w-full h-full flex flex-row justify-center items-center focus:ring-1 font-medium focus:outline-none px-3 py-2 text-sm font-medium text-center rounded-lg text-white bg-blue-700 dark:bg-blue-600 hover:bg-blue-800 dark:hover:bg-blue-700 focus:ring-blue-300 dark:focus:ring-blue-800\\">
  <div class=\\"flex flex-row justify-center items-center\\"><span class=\\"sh-button__prefix mr-2\\"></span><span>Test</span><span class=\\"sh-button__suffix ml-2\\"></span></div>
</button>"
`;
